### PR TITLE
Fix SERIAL_FLOAT_PRECISION compile error

### LINKED ISSUE #20686: [BUG] Compile error because of undefined SERIAL_FLOAT_PRECISION
It seems that latest bugfix has some issue while compiling because SERIAL_FLOAT_PRECISION is not defined (it's commented in configuration_adv.h).
In file temperature.cpp you may find the offending code:
```c++
 SERIAL_PRINT(c, _MIN(SERIAL_FLOAT_PRECISION, 2));
 SERIAL_ECHOPGM(" /");
 SERIAL_PRINT(t, _MIN(SERIAL_FLOAT_PRECISION, 2));
```

I think some sort of default `#define SERIAL_FLOAT_PRECISION 2` must be done since this is a new option and should default to previous behaviours

Cnf not attached because not relevant if HAS_TEMP_SENSOR is true (I think any valid printer configuration has it)

### PR DESCRIPTION
**Pull request by GMagician**
Fix #20686

### CODE REVIEW

File: Marlin/src/core/serial.h [hunk 1]
```diff
@@ -291,7 +291,7 @@ extern uint8_t marlin_debug_flags;
 
 #define SERIAL_ECHO_TERNARY(TF, PRE, ON, OFF, POST) serial_ternary(TF, PSTR(PRE), PSTR(ON), PSTR(OFF), PSTR(POST))
 
-#if SERIAL_FLOAT_PRECISION
+#if SERIAL_FLOAT_PRECISION > 2
```
Comment (qwewer0) [inline Marlin/src/core/serial.h]: Shouldn't it need to work with 0-2 too?
What do you think?

Comment (GMagician) [reply Marlin/src/core/serial.h]: The code I removed tested _MIN(SERIAL_FLOAT_PRECISION, 2).
When missing define it defaults to old 2 digit. Not sure 1 digit has sense

Comment (qwewer0) [reply Marlin/src/core/serial.h]: Forgot about that, ignore me... 😋

Comment (GMagician) [reply Marlin/src/core/serial.h:294]: The move I removed tested _MIN(SERIAL_FLOAT_PRECISION, 2).
When missing define it defaults to old 2 digit. Not sure 1 digit has sense

**Comment by thinkyhead** [comment]
It's probably very unusual for someone to choose 1 digit of precision, but I figure that _MIN is better. I still want to limit temperature output to 100ths of a degree, because anything below that is well into the error margin.

**Pull request merged**